Add basic terraform recipe and localstack test

baierjan opened the pull request:
Reference: https://progress.opensuse.org/issues/118660

baierjan committed:
Add basic terraform recipe

Reference: https://progress.opensuse.org/issues/118660

Martchus reviewed (commented):
I know it is just a draft but adding a bit of documentation wouldn't hurt. Nice that you have already validation tests.

baierjan commented:
> I know it is just a draft but adding a bit of documentation wouldn't hurt. Nice that you have already validation tests.

Yeah, thanks for the note, I also thought about documentation but so far it is just a rewrite of [instructions from the wiki](https://progress.opensuse.org/projects/openqav3/wiki/Wiki#section-71). At the end, I will at least add an explanation how to call the whole thing. But for that, I need to test that first, so more will come.

baierjan committed:
Add Terraform format check and validation into workflow

perlpunk commented on container/terraform/variables.tf:
```diff
@@ -0,0 +1,60 @@
+variable "region" {
+  # preffered region: Frankfurt, Germany
```
```suggestion
  # preferred region: Frankfurt, Germany
```

perlpunk committed:
Use key_name in aws_instance

We need to tell aws which key_name to use.
Then it also shows up in the AWS Console.

What I did:

    ssh-keygen -t rsa -m PEM -f webuidemo.pem
    ssh-keygen -y -f webuidemo.pem >webuidemo.pub

I used the content of webuidemo.pub for the public_key value.

    ssh -i webuidemo.pem -o IdentitiesOnly=yes ec2-user@ip -v

Also I had to use a new image_id as it failed to find the one we had before.

openSUSE-Leap-15-4-v20221105-hvm-ssd-x86_64-5535c495-72d4-4355-b169-54ffa874f849

Check out
https://pint.suse.com/?resource=images&csp=amazon&state=active&region=eu-central-1&search=leap
for images to use

kalikiana committed:
Document deployment via Terraform
